pinot-core/src/main/java/org/apache/pinot/core/plan/FilterPlanNode.java

```diff
@@ -182,7 +189,12 @@ private BaseFilterOperator constructPhysicalOperator(FilterContext filter, int n
         ExpressionContext lhs = predicate.getLhs();
         if (lhs.getType() == ExpressionContext.Type.FUNCTION) {
           if (canApplyH3Index(predicate, lhs.getFunction())) {
-            return new H3IndexFilterOperator(_indexSegment, predicate, numDocs);
+            String functionName = lhs.getFunction().getFunctionName();
+            if (functionName.equals("st_distance") || functionName.equals("stdistance")) {
+              return new H3IndexFilterOperator(_indexSegment, predicate, numDocs);
+            } else {
```

Comment:
nit: better to make condition explicit, and throw exception in else condition

Reply:
sure.